process/sync/baseSync.go

```diff
@@ -95,6 +89,18 @@ type baseBootstrap struct {
 	requestsWithTimeout   uint32
 
 	requestMiniBlocks func(uint32, uint64)
+	getHeaderFromPool func([]byte) (data.HeaderHandler, error)
+
+	getCurrHeader                         func() (data.HeaderHandler, error)
```

Comment:
could we have all these functions in an interface ?

Reply:
done